libraries/Wire/src/Wire.cpp

```diff
@@ -205,6 +287,10 @@ bool TwoWire::begin(int sdaPin, int sclPin, uint32_t frequency)
         started = true;
         goto end;
     }
+    if (!allocateWireBuffer()) {
+        // failed! Error Message already issued
+        goto end;
+    }
     if(!initPins(sdaPin, sclPin)){
         goto end;
     }
```

Comment:
What if `allocateWireBuffer()` succeed, but `initPins(sdaPin, sclPin)` fails and proceed to `goto end`, we should `freeWireBuffer()` in the `end:`

Reply:
Same in SLAVE `TwoWire:begin()`

Reply:
If you look at `allocateWireBuffer()`, you will see that this function has already taken care of it and calls `freeWireBuffer()` before returning.

 `allocateWireBuffer()` will make sure that or it successfully allocated Rx and Tx together or otherwise will will not allocate/free both of them.

Reply:
I think he's saying, if `allocateWireBuffer()` succeeds, but then `initPins()` fails and jumps to end, the buffers will still be allocated and there's no call to free them there

Reply:
Exactly

Reply:
And we should add the `freeWireBuffer()` to the `TwoWire:end()` function as well.

Reply:
Ok, Thanks! Yes, it is right. I'll add the ´freeWireBuffer()`.

Reply:
done